strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/storage/validation/resource/ArtifactOperationsValidator.java

```diff
@@ -106,7 +112,32 @@ public void checkAllowsDeletion(Repository repository)
     {
         if (!repository.allowsDeletion())
         {
-            throw new ArtifactStorageException("Deleting artifacts from " + repository.getType() + " repository is not allowed!");
+            throw new ArtifactStorageException("Deleting artifacts from " + repository.getType() +
+                                               " repository is not allowed!");
+        }
+    }
+
+    public void checkArtifactSize(String storageId,
+                                  String repositoryId,
+                                  MultipartFile uploadedFile)
+            throws ArtifactResolutionException
+    {
+        if (uploadedFile.isEmpty() || uploadedFile.getSize() == 0)
+        {
+            throw new ArtifactResolutionException("Uploaded file is empty.");
+        }
+
+        Repository repository = getConfiguration().getStorage(storageId)
+                                                  .getRepository(repositoryId);
+        long artifactsMaxSize = repository.getArtifactsMaxSize();
+
+        if (artifactsMaxSize > 0)
+        {
+            if (uploadedFile.getSize() > artifactsMaxSize)
+            {
+                throw new ArtifactResolutionException("Uploaded artifact is large for repository " + repositoryId +
```

Comment:
Could you please change the error message to `"The size of the artifact exceeds the maximum size accepted by this repository (${size}/${artifactMaxSize}."`?